### ISSUE TITLE
Enhancement: [no-floating-promises] flag `.filter(async)` 

### ISSUE THREAD

**Issue by yelly** [typescript-eslint/typescript-eslint#10441]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/no-floating-promises/

### Description

I propose that the `no-floating-promises` rule should fail when passing an async function to `Array.prototype.filter()` since this is probably always a bug (the filter does nothing).

### Fail

```TypeScript
[1, 2, 3, 4].filter(async x => await new Promise(() => x % 2 == 0))
```

### Pass

```TypeScript
[1, 2, 3, 4].filter(x => x % 2 == 0)
```

### Additional Info

This is similar to #5958.

**Comment by yelly** [comment]
My bad, didn't have my playground configured correctly...
Sorry for the spam.
